src/EntraModuleBuilder.ps1

```diff
@@ -193,6 +193,49 @@ Set-StrictMode -Version 5
             return $subModules.Name
         }
     }
+  [string[]] GetSubModuleManifestFiles([string] $Module, [string]$DirectoryPath) {
+        # Check if the directory exists
+        # Define the pattern for matching submodule files
+        $pattern = if ($module -like "Microsoft.Graph.Entra.Beta.*") {
+            "Microsoft.Graph.Entra.Beta.*.psd1"
+        } else {
+            "Microsoft.Graph.Entra.*.psd1"
+        }
+
+        if (-Not (Test-Path -Path $DirectoryPath)) {
+            Log-Message "[EntraModuleBuilder]: Directory does not exist: $directoryPath" -ForegroundColor Red
+            return $null # Return null if directory does not exist
+        }
+
+     [string[]] GetSubModuleFileNames([string] $Module, [string]$DirectoryPath) {
+        # Check if the directory exists
+        # Define the pattern for matching submodule files
+        $pattern = if ($module -like "Microsoft.Graph.Entra.Beta.*") {
+            "Microsoft.Graph.Entra.Beta.*.psd1"
+        } else {
+            "Microsoft.Graph.Entra.*.psd1"
+        }
+
+        if (-Not (Test-Path -Path $DirectoryPath)) {
+            Log-Message "[EntraModuleBuilder]: Directory does not exist: $directoryPath" -ForegroundColor Red
+            return $null # Return null if directory does not exist
+        }
+
+        # Get all .psm1 files in the specified directory
+        $subModules = Get-ChildItem -Path $DirectoryPath -Filter $pattern -File
+
+        # Check if any .psm1 files were found
+        if ($subModules.Count -eq 0) {
+            Log-Message "[EntraModuleBuilder]: No .psd1 files found in the directory: $directoryPath" -Level 'INFO'
+            return @() # Return an empty array if no files are found
+        } else {
+
+            # Return the names of the .psd1 files
+            return $subModules | ForEach-Object { [System.IO.Path]::GetFileNameWithoutExtension($_.Name) }
```

Comment:
`return` ends the pipeline, so there could be more flexibility in how this script is used if you remove the `return` statement from this line but leave the rest.